storage-proofs-update/src/vanilla.rs

```diff
@@ -252,6 +254,90 @@ where
     }
 }
 
+// Computes all `2^h` rho values for the given `phi` in the given range. Each rho corresponds to
+// a `high` value, where `high` is the `h` high bits of a node-index.
+#[derive(Debug)]
+pub struct Rhos {
+    rhos: HashMap<usize, Fr>,
+    /// The amount of right shift that is needed to get the `h` high bits of a node.
+    bits_shr: usize,
+}
+
+impl Rhos {
+    /// Generate the `rho`s for a certain number of nodes.
+    ///
+    /// Those are used for encoding. The `nodes_count` is the total number of nodes of the sector.
+    pub fn new(phi: &TreeRDomain, nodes_count: usize) -> Self {
```

Comment:
Can these constructors accept an `h` parameter, which gets passed to `Self::calc_bits_shr` rather than forcing `h = h_default(sector_nodes)` (and call `h_default` at the api level and pass it to the constructors)?